### ISSUE TITLE
Parameter for displaying dynamic content

### ISSUE THREAD

**Issue by SimonSteinberger** [thephuse/vanilla-modal#4]
Really a wonderful plugin! Thanks :+1: 

It would be great if an HTML content string could be used for instantiating a modal window.

E.g. like this:

    var modal = new VanillaModal({ content: '<h2>Hello World!</h2>' });
    modal.open();

That makes it easier to use the plugin with AJAX retrieved content, which is often useful when dealing with modal dialogues.

As an unimportant side note concerning "Doesn't work on any prior versions of IE (read: they're un-tested, but feel free to test)" => Yep, doesn't work on IE8 and below.

Another unimportant side note: I think it would be cleaner to use classes instead of the "rel" attribute as default action attributes. The "rel" attribute isn't meant to be used for this and doesn't pass validation, although it technically doesn't hurt.

**Comment by benceg** [comment]
Hi @SimonSteinberger,

Thanks for the kudos and comments!

To address your first suggestion: string injection is certainly on the cards, although it wasn't in the 1.x release due to the implications it has on performance as opposed to using native DOM methods to find and insert content. But what with the ES 2015 template string syntax having been finalized, it'd seem silly not to throw it in.

As to the matter of `[rel]` being the default selector: this has come up before, and I quite agree, but I'm not going to change the default while we're at v1.x just yet, in order to maintain backwards compatibility (especially for Browserify and Webpack users). If you'd prefer to use classes, you can set them on instantiation:

```js
var modal = new VanillaModal({
  open : '.open',
  close : '.close'
});
```

...and that will help you keep the rest of your codebase standards-compliant. Glad you're enjoying using the script! If you'd like to let us know where you're using it, we can include the address in an As Seen On section of the readme - which is itself long overdue!

(I'm leaving this issue open as I'll be addressing both of your points  in future.)

**Comment by SimonSteinberger** [comment]
Thanks for the reply :-) Yeah, I noticed the closed issue about the rel tag after I posted this here.

It's all not a big issue. It's easy to write a wrapper for the currently missing string injection. I'll let you know as soon as we're done with our project.

**Comment by benceg** [comment]
Hi @SimonSteinberger,

You'll be pleased to know that both of the above issues are being addressed with release 2.0.0, on which I'm currently working. It's a large overhaul with a few breaking changes – hence the semver leap. It'll be in circulation shortly via NPM and GitHub, and I hope you'll enjoy using it! I'll close this ticket for now, thanks again for the points you've raised here.
